pkg/querier/querier.go

```diff
@@ -99,7 +101,7 @@ type responseFromIngesters struct {
 // forAllIngesters runs f, in parallel, for all ingesters
 // TODO taken from Cortex, see if we can refactor out an usable interface.
 func (q *Querier) forAllIngesters(ctx context.Context, f func(logproto.QuerierClient) (interface{}, error)) ([]responseFromIngesters, error) {
-	replicationSet, err := q.ring.GetAll()
+	replicationSet, err := q.ring.GetAll(ring.Read)
```

Comment:
I guess in all place `GetAll` should be `ring.Read`. Do we need `ring.Write` in any place?

Reply:
/cc @pstibrany

Reply:
Leaving ingesters still have data, and can be used for querying.

For Pending, there is unfinished discussion about that in https://github.com/cortexproject/cortex/pull/1866.

Reply:
Ok let's keep it like this then.`ring.Read`